pysat/tests/test_instrument.py

```diff
@@ -48,49 +48,125 @@ def test_basic_instrument_load(self):
         data)."""
         self.testInst.load(self.ref_time.year, self.ref_doy)
         self.out = self.testInst.index[0]
+        assert (self.out == self.ref_time)
         self.out = dt.datetime(self.out.year, self.out.month, self.out.day)
+        assert (self.out == self.testInst.date)
+
+    def test_basic_instrument_load_two_days(self):
+        """Test if the correct day is being loaded (checking object date and
+        data)."""
+        self.testInst.load(self.ref_time.year, self.ref_doy,
+                           self.ref_time.year, self.ref_doy + 2)
+        self.out = self.testInst.index[0]
         assert (self.out == self.ref_time)
+        self.out = dt.datetime(self.out.year, self.out.month, self.out.day)
         assert (self.out == self.testInst.date)
+        self.out = self.testInst.index[-1]
+        assert (self.out >= self.ref_time + pds.DateOffset(days=1))
+        assert (self.out <= self.ref_time + pds.DateOffset(days=2))
 
     def test_basic_instrument_bad_keyword(self):
         """Checks for error when instantiating with bad load_rtn keywords"""
         with pytest.raises(ValueError):
             pysat.Instrument(platform=self.testInst.platform,
-                             name=self.testInst.name, inst_id='10',
+                             name=self.testInst.name, num_samples=10,
                              clean_level='clean',
                              unsupported_keyword_yeah=True)
 
     def test_basic_instrument_load_yr_no_doy(self):
+        """Ensure doy required if yr present"""
         with pytest.raises(TypeError):
             self.testInst.load(self.ref_time.year)
 
+    def test_basic_instrument_load_yr_no_end_doy(self):
+        """Ensure end_doy required if end_yr present"""
+        with pytest.raises(ValueError):
```

Comment:
Here and below, recommend testing value error message to ensure correct ValueError is triggered.

Reply:
Adding testing for error messages (not this particular method) revealed an issue with the pysat_testing2D instrument when num_samples < 50. Thanks.